test/gtest/unit_ActivationDescriptor.cpp

```diff
@@ -0,0 +1,532 @@
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ *******************************************************************************/
+
+#include <algorithm>
+#include <future>
+
+#include <miopen/activ.hpp>
+// \todo This should be removed when the testing infrastructure is improved
+#include <miopen/env.hpp>
+#include <miopen/miopen.h>
+#include <miopen/handle.hpp>
+
+#include "activ_common.hpp"
+#include "gtest_common.hpp"
+
+#include "../network_data.hpp"
+#include "../random.hpp"
+#include "../tensor_holder.hpp"
+#include "../verify.hpp"
+
+// \todo This should be removed when the testing infrastructure is improved
+MIOPEN_DECLARE_ENV_VAR_STR(MIOPEN_TEST_FLOAT_ARG)
+MIOPEN_DECLARE_ENV_VAR_BOOL(MIOPEN_TEST_ALL)
+
+namespace {
+
+std::set<std::vector<int>> GetTestCases(bool full)
+{
+    if(full)
+        return get_inputs(0);
+    else
+        return {{1, 16, 8, 8}};
+}
+
+auto GetAllActivationModes()
+{
+    return std::vector<miopenActivationMode_t>{
+        miopenActivationPASTHRU,
+        miopenActivationLOGISTIC,
+        miopenActivationTANH,
+        miopenActivationRELU,
+        miopenActivationSOFTRELU,
+        miopenActivationABS,
+        miopenActivationPOWER,
+        miopenActivationCLIPPEDRELU,
+        miopenActivationLEAKYRELU,
+        miopenActivationELU,
+    };
+}
+
+Gpu GetSupportedDevices() { return Gpu::All; }
+
+template <class T>
+class UnitTestActivationDescriptor
+    : public ::testing::TestWithParam<std::tuple<Gpu, miopenActivationMode_t, std::vector<int>>>
+{
+public:
+    void RunTest()
+    {
+        miopenActivationMode_t mode;
+        std::vector<int> input_dims;
+        std::tie(std::ignore, mode, input_dims) = GetParam();
+
+        const double alpha = 0.95;
+        const double beta  = 2.3;
+        const double gamma = 3.4;
+        const auto desc    = miopen::ActivationDescriptor{mode, alpha, beta, gamma};
+
+        auto x_tensor      = tensor<T>{input_dims};
+        auto y_tensor_gpu  = tensor<T>{input_dims};
+        auto y_tensor_cpu  = tensor<T>{input_dims};
+        auto dx_tensor_gpu = tensor<T>{input_dims};
+        auto dy_tensor     = tensor<T>{input_dims};
+        auto dx_tensor_cpu = tensor<T>{input_dims};
+
+        auto x_data_ready      = std::async(std::launch::async, [&]() { GenData_x(x_tensor); });
+        auto y_data_gpu_ready  = std::async(std::launch::async, [&]() { ZeroData(y_tensor_gpu); });
+        auto dx_data_gpu_ready = std::async(std::launch::async, [&]() { ZeroData(dx_tensor_gpu); });
+        auto dy_data_ready     = std::async(std::launch::async, [&]() { GenData_dy(dy_tensor); });
+
+        x_data_ready.wait();
+        y_data_gpu_ready.wait();
+        dx_data_gpu_ready.wait();
+        dy_data_ready.wait();
```

Comment:
If you pass it right into `RunCpu`/`RunGpu` you can postpone `dx`/`dy` wait till backward pass. (just an idea for already complex async code).